fix(onboarding): Modify useCurrentProjectState to not require an additional rerender

malwilley opened the pull request:
Am updating some onboarding components and noticed that this hook will always require one rerender to return the correct project. Small refactor to find the default project on initial render if the projects are already available.

malwilley committed:
fix(onboarding): Modify useCurrentProjectState to not require an additional rerender

ryan953 commented on static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx:
```diff
@@ -37,26 +36,23 @@ function useCurrentProjectState({
-  useEffect(() => {
+  const getDefaultCurrentProject = useCallback((): Project | undefined => {
     if (!isActive) {
-      setCurrentProject(undefined);
```
nice! in hindsight calling set-state inside a useEffect isn't the best way.

michellewzhang commented:
i think this breaks the replay onboarding's (maybe other onboardings, i didn't try) current behavior of selecting the correct project when a diff project is selected from the project dropdown


https://github.com/user-attachments/assets/ef6b08fb-a61b-4336-a2c8-41732079455a

malwilley committed:
Fix for selecting from page filters

malwilley commented:
> i think this breaks the replay onboarding's (maybe other onboardings, i didn't try) current behavior of selecting the correct project when a diff project is selected from the project dropdown

@michellewzhang you're right thanks for catching! Fixed and added a test for that behavior

michellewzhang reviewed (approved):
looks good! thanks for fixing
